src/components/config_parser.cpp

```diff
@@ -0,0 +1,273 @@
+#include <fstream>
+#include <algorithm>
+
+#include "components/config_parser.hpp"
+
+POLYBAR_NS
+
+config_parser::config_parser(const logger& logger, string&& file, string&& bar)
+  : m_log(logger),
+  m_config(file_util::expand(move(file))),
```

Comment:
```suggestion
  m_config(file_util::expand(file)),
```
This move is useless since file_util::expand takes a const ref.